controllers/controllers/cluster_controller.go

```diff
@@ -32,32 +38,70 @@ const (
 
 // ClusterReconciler reconciles a Cluster object
 type ClusterReconciler struct {
-	client    client.Client
-	log       logr.Logger
-	validator *vsphere.Validator
-	defaulter *vsphere.Defaulter
-	tracker   *remote.ClusterCacheTracker
+	client                  client.Client
+	log                     logr.Logger
+	validator               *vsphere.Validator
+	defaulter               *vsphere.Defaulter
+	tracker                 *remote.ClusterCacheTracker
+	buildProviderReconciler BuildProviderReconciler
 }
 
-func NewClusterReconciler(client client.Client, log logr.Logger, scheme *runtime.Scheme, govc *executables.Govc, tracker *remote.ClusterCacheTracker) *ClusterReconciler {
+// TODO: this is not ideal and will need a refactor. I will follow up but for now this
+// allows us to decouple the cluster reconciler main logic from provider specific logic
+type BuildProviderReconciler func(datacenterKind string, client client.Client, log logr.Logger, validator *vsphere.Validator, defaulter *vsphere.Defaulter, tracker *remote.ClusterCacheTracker) (clusters.ProviderClusterReconciler, error)
```

Comment:
I will come up with a better way of doing this, it's just a minimal change to allow me to dummy/mock the providers specific reconciler so I can test the cluster reconciler logic alone

I was thinking about having a `ClusterProviderReconcilerRegistry` that returns the proper reconciler based on the `Cluster`. This would be built as any other dependency and injected in the constructor. But TBD until I get there